### ISSUE TITLE
Behavior of control + click changed in 0.9.214?

### ISSUE THREAD

**Issue by Gert-dev** [gnumdk/lollypop#824]
-Lollypop version: 0.9.217
-Gtk Version: 3.22
-Linux distribution: Arch Linux

-Describe your bug here:

Lollypop has the functionality "Stop after current", which is also accessible by using "Control + click" to toggle it automatically. As my original feature request to make this switchable by default (to have "Control + click" continue playing and normal click stop after current) was rejected, I've been manually patching Lollypop to reverse the behavior ever since, which is no big deal since it's just a single line.

However, in version 0.9.214 this behavior was unexpectedly replaced with an entirely different action, see also [1] and [2]. At first I thought Lollypop was bugging out and reverted to 0.9.213, but looking at the source the behavior now makes sense.

Is there a reason control + click now suddenly skips the entire album instead of enabling "stop after current" for the current track? I don't mind these changes if there is a good reason for them (even though it breaks my use case), but would have appreciated a note in the changelog about it.

Thanks in advance

[1] https://github.com/gnumdk/lollypop/blob/0.9.213/src/widgets_album.py#L1084
[2] https://github.com/gnumdk/lollypop/blob/0.9.214/src/widgets_album.py#L1100

**Comment by gnumdk** [comment]
New code since 0.9.214 is a bug, my idea was to remove this hidden feature...

But, why do you want it to be the default? 
#809

**Comment by gnumdk** [comment]
By the way, I was not really happy with #809 fix...

And with 0.9.218 to come, you can stop patching lollypop ;)

**Comment by Gert-dev** [comment]
That's great to hear!

The old behavior was indeed to do "stop after current" once and then revert to "continue playing" after the current song finished (or another song was started by clicking it). Fo

r this reason I patched in a "not" operator so every click behaved like a control + click and I always got "stop after current" by clicking a song. So whether it automatically restored the behavior afterwards or not didn't really affect me since I was always overriding it back to "stop after current" anyway :-).

I understand that Lollypop is an album player and you usually want to play an entire album (i.e. not use "stop after current") and I do use it for this purpose often, but usually I'm just looking up songs I like and click to play them. I could also make a playlist for this purpose but I don't always know beforehand what song I want to listen to next.

I still find it a bit strange, perhaps because I'm a bit indoctrinated by other music players in the past :-), that clicking a single song from an album doesn't stop automatically after that song, but instead continues to play the next song. I'll try to explain a bit more: it feels like, at least in most cases, I either want to...
1. ... play an entire artist's collection, perhaps by clicking the artist cover or artist name, without stopping automatically after any song.
2. ... play an entire album without stopping automatically at any of the songs, in which case you can click the big play button on the album cover.
3. ... play a single song from a single album, in which case you click the song you want to play and it also stops after that song.

In other words, if I click the 9th song out of 10, it feels a bit awkward that the 10th song is automatically played after the 9th one finishes, as I'm not in "album playing mode" (if I was, I would be starting at the first song and playing the entire album).

Of course, this is just my opinion and reasoning, since you so kindly asked ;-).

**Comment by gnumdk** [comment]
>Of course, this is just my opinion and reasoning, since you so kindly asked ;-).
So my fix is not working for you :-( And in any CD player, starting at song 9th will play song 10th

**Comment by gnumdk** [comment]
Ok, rework all the code as you're suggestion was not that bad ;)

So, simplified repeat menu: continue, repeat track, not repeat

More logical from a user point of view.

**Comment by Gert-dev** [comment]
About the CD player, that's very true. I also realized that most other players will also just continue playing, but also provide an option to change the behavior and always stop after the current track. Lollypop already supports this, of course, but currently it resets after the current track stops playing, which I why I was patching the click action to automatically set it each time.

In the Git version which I just built, the control + click option now seems to be gone. This isn't really a problem for me directly as I wasn't using control + click anyway. I'd be happy to continue patching Lollypop to automatically enable stop after current in the `_on_activate` method, could you give me a quick pointer as to how I can enable it programmatically? It used to be `Lp().player.set_next_context(NextContext.STOP_TRACK)`, but it seems that some code has moved?

**Comment by Gert-dev** [comment]
Oops, it seems like commit 969d4604832d41231595979c1468d05113f367a3 broke it again, I now have the following behavior:
* "Continue playing" keeps playing and doesn't stop after current (:+1:)
* "Repeat song" doesn't stop after current, but repeats the current song (:+1).
* "Don't repeat" also just keeps playing and doesn't stop after current.

**Comment by gnumdk** [comment]
Hmm, "Don't repeat" stop after current if you didn't add album to playlist (play button/add button on cover).

Will not handle a special case on this, if you click on a file, lollypop stop after this file only if you didn't set an album to play...

**Comment by Gert-dev** [comment]
Yes, that was indeed it. I was not aware that the album got "queued" when you click to play it (via the cover), so I assumed stop after current wasn't working. In fact it is as you say, it only works when no album is queued, so if I remove the album from the queue first, then it works.

I don't mind this behavior and I think it would even be awkward if the album disappeared from the queue if you clicked a single song, but would it perhaps be a good idea to somehow convey this behavior to the user? In other words, I now understand how it works and how I can get it to work, but I can imagine new users that select "Don't repeat" may be wondering why "it's not working" whilst they may have played an album before and have no idea that it's still queued and it's influencing the behavior?
